### ISSUE TITLE
MD060: Condider allowing opt-out of CJK character double-width behavior

### ISSUE THREAD

**Issue by EuropeanDeveloper** [DavidAnson/markdownlint#2034]
## Description
I’m getting a warning in VS Code for the following situation:

```markdown
| Column 1 | Column 2                                                          |
|----------|-------------------------------------------------------------------|
|          | ⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛ |
|          | ⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛ |
|          | ⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛ |
|          | ⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛ |
|          | ⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛ |
|          | ⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛ |
|          | ⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛ |
|          | ⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛ |
|          | ⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛ |
|          | ⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛⬛ |
```

<img width="864" height="233" alt="Image" src="https://github.com/user-attachments/assets/aa78191e-2cf6-408d-a584-12e9a308f995" />

As per [MD060](https://github.com/DavidAnson/markdownlint/blob/main/doc/md060.md):
> **Note**: Pipe alignment for the `aligned` style is based on visual appearance
> and not character count. Because editors typically render emoji and
> CJK characters at *twice* the width of
> Latin characters, this rule takes that into account for tables
> using the `aligned` style. The following table is correctly formatted and will
> appear aligned in most editors and monospaced fonts:
> 
> ```markdown
> | Response | Emoji |
> | -------- | ----- |
> | Yes      | ✅    |
> | No       | ❎    |
> ```

I believe the columns are aligned correctly.

When I open the same file in Notepad++, it look like this:

<img width="641" height="203" alt="Image" src="https://github.com/user-attachments/assets/7dfa7f51-e902-4e8f-a101-1e9e2adbeb52" />

If I adjust the last column to:

<img width="537" height="205" alt="Image" src="https://github.com/user-attachments/assets/a5bb9695-00a7-4df9-aedf-55774e3798bb" />

the warning in VS Code disappears, but the table no longer looks properly aligned:

<img width="619" height="232" alt="Image" src="https://github.com/user-attachments/assets/2ab1fb51-8197-4269-94fb-424c325d04c1" />

I don’t think enforcing visual alignment is the right approach, since emojis are rendered differently in each editor. It also varies by theme – for example, Dark mode in Notepad++ looks different from Light mode:

<img width="470" height="184" alt="Image" src="https://github.com/user-attachments/assets/f5b17cd5-0817-4ec5-90c3-c9d51fceed4e" />

Because of these differences, relying on visual alignment seems unreliable. It would be useful to have an option to configure this rule to use character count instead of visual alignment, ensuring consistent behavior across editors.

## Environment
```
markdownlint 0.61.2
VS Code 1.115.0 (user setup)
```

**Comment by DavidAnson** [comment]
The initial implementation of this rule used character count, but that was problematic for Chinese/Japanese/Korean characters: https://github.com/DavidAnson/markdownlint/issues/1869

The current implementation handles CJK as expected by CJK users and is generally/mostly right for emoji as well.

As you point out, editors can render characters however they choose and therefore this rule cannot satisfy every circumstance. Adding an option to use character count would not magically align your scenario (the block emoji are wider than the dashes) and would cause problems for any of your documents containing CJK characters.

I think the proposal introduces a different set of problems without addressing the underlying issue. If this rule is not helpful for your scenario, I recommend disabling it.

**Comment by EuropeanDeveloper** [comment]
I'm sorry, I could have explained it more clearly. I found the rule very useful, especially when working with plain (Western) text. The current implementation is unfortunate to my mind – I use your VS Code extension (I know it's a different repository) and this rule is not fitted even for rendering emojis in this editor. I would expect it to at least fit properly in VS Code.

I would be happier if I could configure the rule to use character count. That way, it would behave consistently across different editors. I could align the tables by line length and accept that emojis may render differently, even if columns may not always perfectly align.
